crates/ruff/src/commands/rule.rs

```diff
@@ -18,26 +19,25 @@ struct Explanation<'a> {
     linter: &'a str,
     summary: &'a str,
     message_formats: &'a [&'a str],
-    fix: String,
     #[expect(clippy::struct_field_names)]
     explanation: Option<&'a str>,
-    preview: bool,
+    fix: FixAvailability,
+    group: RuleGroup,
 }
 
 impl<'a> Explanation<'a> {
     fn from_rule(rule: &'a Rule) -> Self {
         let code = rule.noqa_code().to_string();
         let (linter, _) = Linter::parse_code(&code).unwrap();
-        let fix = rule.fixable().to_string();
         Self {
             name: rule.name().as_str(),
             code,
             linter: linter.name(),
             summary: rule.message_formats()[0],
             message_formats: rule.message_formats(),
-            fix,
             explanation: rule.explanation(),
-            preview: rule.is_preview(),
+            fix: rule.fixable(),
```

Comment:
I agree that this is an improvement, but I'm not sure if it's worth going through the complications of adding a preview flag (which might even be more annoying for upstream tools than an outright breaking change because they now need to handle both formats if the user set `preview = true` in their configuration). 

I'm inclined to just leave it as is OR introduce a new field `fix_availability` and deprecate the old field.

Reply:
I thought having the enum values instead of entire sentences like `"Fix is sometimes available."` would allow users to parse the json more easily, which was the point raised by the issue
We could also leave both fields, even if this would mean some duplication...

Not sure what's best... Let me know what you prefer :)

Reply:
Yes, we need to restore any breaking change.